src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotManager.java

```diff
@@ -193,4 +205,52 @@ protected final void withLock(Runnable action) {
       bufferLock.unlock();
     }
   }
+
+  /**
+   * Clear all data of snapshot for the provided feed id
+   */
+  public void clearBuffer(String feedId) {
+    buffer.clear(feedId);
+  }
+
+  /**
+   * Update the TripTimes of one Trip in a Timetable of a TripPattern. If the Trip of the TripTimes
+   * does not exist yet in the Timetable, add it. This method will make a protective copy
+   * of the Timetable if such a copy has not already been made while building up this snapshot,
+   * handling both cases where patterns were pre-existing in static data or created by realtime data.
+   *
+   * @param serviceDate service day for which this update is valid
+   * @return whether the update was actually applied
+   */
+  public Result<UpdateSuccess, UpdateError> updateBuffer(
+    TripPattern pattern,
+    TripTimes tripTimes,
+    LocalDate serviceDate
+  ) {
+    return buffer.update(pattern, tripTimes, serviceDate);
+  }
+
+  /**
+   * Removes the latest added trip pattern from the cache. This should be done when removing the
+   * trip times from the timetable the trip has been added to.
+   */
+  public void removeLastAddedTripPattern(FeedScopedId id, LocalDate serviceDate) {
+    buffer.removeLastAddedTripPattern(id, serviceDate);
+  }
+
+  public void removeRealtimeUpdatedTripTimes(
```

Comment:
It would be nice with javadoc on this one

Reply:
Ok, if that's already being done in another PR there is no point in doing it here.